src/ncp/ncp_base.hpp

```diff
@@ -532,11 +590,35 @@ class NcpBase
     static NcpBase        *sNcpInstance;
     static spinel_status_t ThreadErrorToSpinelStatus(otError aError);
     static uint8_t         LinkFlagsToFlagByte(bool aRxOnWhenIdle, bool aDeviceType, bool aNetworkData);
-    Instance              *mInstance;
-    Spinel::Buffer         mTxFrameBuffer;
-    Spinel::Encoder        mEncoder;
-    Spinel::Decoder        mDecoder;
-    bool                   mHostPowerStateInProgress;
+
+    /**
+     * Returns an IID for the given instance
+     *
+     * Returned IID is an integer value that must be shifted by SPINEL_HEADER_IID_SHIFT before putting into spinel
+     * header. If multipan interface is not enabled or build is not for RCP IID=0 is returned. If nullptr is passed it
+     * matches broadcast IID in current implementation. Broadcast IID is also returned in case no match was found.
+     *
+     * @param[in] aInstance  Instance pointer to match with IID
+     *
+     * @returns Spinel Interface Identifier to use for communication for this instance
+     *
+     */
+    uint8_t InstanceToIid(Instance *aInstance);
```

Comment:
Can we make `InstanceToIid` and `IidToInstance` APIs to be `public` members? So that it can be used in PAL rather than maintaining one more copy of `mInstances`. Then, we can have something like 
```
extern "C" otInstance* otIidToInstance(uint8_t aIid)
{
    return ot::Ncp::NcpBase::GetNcpInstance()->IidToInstance(aIid);
}

extern "C" uint8_t otInstanceToIid(otInstance *aInstance)
{
    return ot::Ncp::NcpBase::GetNcpInstance()->InstanceToIid(static_cast<ot::Instance *>(aInstance));
}
```